scripts/monorepo/bump-all-updated-packages/bump-package-version.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+
+const {writeFileSync, readFileSync} = require('fs');
+const path = require('path');
+
+const forEachPackage = require('../for-each-package');
+
+const ROOT_LOCATION = path.join(__dirname, '..', '..', '..');
+const TEMPLATE_LOCATION = path.join(ROOT_LOCATION, 'template');
+const REPO_CONFIG_LOCATION = path.join(ROOT_LOCATION, 'repo-config');
+
+const readJSONFile = packageAbsolutePath =>
+  JSON.parse(readFileSync(packageAbsolutePath));
+
+const getIncrementedVersion = (version, increment) =>
+  version
+    .split('.')
+    .map((token, index) => {
+      const indexOfVersionToIncrement = increment === 'minor' ? 1 : 2;
+
+      return index === indexOfVersionToIncrement
+        ? parseInt(token, 10) + 1
+        : token;
+    })
+    .join('.');
+
+const updatePackageVersionInPackageManifest = (
+  packageAbsolutePath,
+  updatedPackageName,
+  updatedPackageVersion,
+) => {
+  const packageManifestPath = path.join(packageAbsolutePath, 'package.json');
+  const packageManifest = readJSONFile(packageManifestPath);
+
+  const dependencyVersion = packageManifest.dependencies?.[updatedPackageName];
+  if (dependencyVersion && dependencyVersion !== '*') {
+    const updatedDependencyVersion = dependencyVersion.startsWith('^')
+      ? `^${updatedPackageVersion}`
+      : updatedPackageVersion;
+    const updatedPackageManifest = {
+      ...packageManifest,
+      dependencies: {
+        ...packageManifest.dependencies,
+        [updatedPackageName]: updatedDependencyVersion,
+      },
+    };
+
+    writeFileSync(
+      packageManifestPath,
+      JSON.stringify(updatedPackageManifest, null, 2) + '\r\n',
```

Comment:
Won't this cause mixed newlines depending on which platform you're running this script on?

Reply:
Yes, probably. Will try to find how to avoid using it, because every package manifest that we have has an empty line in the end and using just `JSON.stringify` produces a content without empty line. Maybe I need to specify something to make it work

Reply:
I've replaced this with `JSON.stringify(updatedPackageManifest, null, 2) + '\n'`, removed `\r`, should work the same on different platforms